scripts/zones/Southern_San_dOria/npcs/Kipopo.lua

```diff
@@ -6,18 +6,41 @@
 -----------------------------------
 require("scripts/globals/status");
 require("scripts/globals/crafting");
+require("scripts/globals/quests");
 local ID = require("scripts/zones/Southern_San_dOria/IDs");
 -----------------------------------
 
 function onTrade(player,npc,trade)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+
+    if (player:hasKeyItem(dsp.ki.TORN_PATCHES_OF_LEATHER) and sayItWithAHandbagCS == 2 and trade:getItemCount() == 3) then
+        if (trade:hasItemQty(2012,1) and trade:hasItemQty(850,1) and trade:hasItemQty(816,1)) then
```

Comment:
[`npcutil.tradeHas` can shorten this](https://github.com/DarkstarProject/darkstar/blob/master/scripts/globals/npc_util.lua#L347-L360)

Reply:
O M G🙀! there's documentation 👍 Totally awesome, fixing now 😸